pkg/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,185 @@
+// and then, according to the queue priority, sequentially retrieving all RBs from the queues.
+// If each RB meets certain conditions,it will be placed in the queue
+// and subsequently updated with their Suspend set to false.
+func (dispatcher *Dispatcher) dispatch(ssn *dispatcherframework.Session) {
+	klog.V(5).Infof("Dispatcher start running...")
+	defer klog.V(5).Infof("Dispatcher end running...")
+
+	ss := ssn.Snapshot
+	queues := util.NewPriorityQueue(ssn.QueueInfoOrderFn)
+	resourceBindingMap := map[string]*util.PriorityQueue{}
+	// The counts for logs.
+	enqueueResourceBindingCount := 0
+	dispatchResourceBindingCount := 0
+
+	// Collect the workloads to the queue map.
+	// For now, the `workload` includes Deployment, volcano-job and Pod only.
+	// Because only the three resources will create PodGroup by controllers.
+	for _, rbi := range ss.ResourceBindingInfos {
+		rb := rbi.ResourceBinding
+
+		// Check if its Suspended, dispatcher cares the suspended rbi only.
+		if rbi.DispatchStatus != api.Suspended {
+			continue
+		}
+
+		// If its workload but without PodGroup, skip it.
+		// Only workload ResourceBinding will be suspend and add to the dispatcher cache.
+		if rbi.PodGroup == nil {
+			klog.Errorf("ResourceBinding <%s/%s> is a workload but has no PodGroup, stop dispatching and enqueue.",
+				rb.Namespace, rb.Name)
+			continue
+		}
+
+		// Get the workload's queue name, it may be a nil.
+		rbiQueueName := ssn.GetResourceBindingInfoQueue(rbi)
+		resource := rb.Spec.Resource
+
+		// Check if the queue set in the map.
+		if rbiPriorityQueue, found := resourceBindingMap[rbiQueueName]; found {
+			// Add this workload to the queue.
+			rbiPriorityQueue.Push(rbi)
+		} else {
+			// This queue didn't set in the map, we should check if the queue exists first, then add it to the map.
+			if queue, found := ss.QueueInfos[rbiQueueName]; found {
+				klog.V(5).Infof("Added Queue <%s> for ResourceBinding <%s/%s>.",
+					rbiQueueName, rb.Namespace, rb.Name)
+				// Create the priority queue for ResourceBindings, and push it.
+				resourceBindingMap[rbiQueueName] = util.NewPriorityQueue(ssn.ResourceBindingInfoOrderFn)
+				resourceBindingMap[rbiQueueName].Push(rbi)
+
+				queues.Push(queue)
+				enqueueResourceBindingCount++
+			} else {
+				// We cant find this queue in the cache snapshot, skip it.
+				klog.V(3).Infof("Resource %s <%s/%s> Queue <%s> not found, skip dispatching.",
+					resource.Kind, resource.Namespace, resource.Name, rbiQueueName)
+				continue
+			}
+		}
+	}
+
+	klog.V(5).Infof("Success enqueue <%d> ResourceBindingInfos and <%d> Queues, start dispatching now...",
+		enqueueResourceBindingCount, len(resourceBindingMap))
+
+	for {
+		// Finish dispatching when all the queues dispatch done.
+		if queues.Empty() {
+			break
+		}
+
+		queue := queues.Pop().(*schedulingapi.QueueInfo)
+		resourceBindingsQueue := resourceBindingMap[queue.Name]
+
+		// Get all the ResourceBindingInfos from the priority queue.
```

Comment:
It is logical that this place should no longer take rb from the current queue after the queue is saturated, if the judgment of queue saturation has not yet been implemented, you can cyclically take an rb from each queue to ensure the fairness of the different queues dispatch rb.